.gitmodules

```diff
@@ -1,3 +1,22 @@
 [submodule "third_party/prometheus-cpp"]
 	path = third_party/prometheus-cpp
 	url = https://github.com/jupp0r/prometheus-cpp.git
+
+[submodule "tools/vcpkg"]
+	path = tools/vcpkg
+	url = https://github.com/Microsoft/vcpkg
+	branch = master
```

Comment:
We are good here. We basically pin submodule to commit id. Every time when we need to upgrade it, it's gonna be zero-diff PR that shows to what new commit ID we are moving on to... This will only be used for CMake, as Bazel manages all deps separately. The only use-case where we need this, is Prometheus library for the Metrics exporter --- right now Bazel build expects it to be there, checked out in source tree. I think we'll eventually need to fix this for Bazel, to fetch from git instead... Then for Bazel builds there would be no need to even clone the submodules: it could be 'shallow' clone without submodules.

Reply:
We will handle it nicely: if our dependency publishes their release on the master, we refer to their master Commit ID that is on of their release Tags. If they don't publish on the master, we will use their release branch. In most cases we use the master. GSL, for example, publishes their releases with one branch - master only.